### PR TITLE
feat(http): add http service for tasks

### PR DESCRIPTION
**Pull request by jademcgough**

### CODE REVIEW

File: http/task_service.go [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+
+type getLogsRequest struct {
+	filter platform.LogFilter
+}
+
+func decodeGetLogsRequest(ctx context.Context, r *http.Request) (*getLogsRequest, error) {
+	qp := r.URL.Query()
+	req := &getLogsRequest{}
+
+	if id := qp.Get("task"); id != "" {
```
Comment (goller) [inline http/task_service.go]: I think this should be `tid`

File: http/task_service.go [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+		req.filter.Task = &platform.ID{}
+		if err := req.filter.Task.DecodeFromString(id); err != nil {
+			return nil, err
+		}
+	}
+
+	if id := qp.Get("run"); id != "" {
```
Comment (goller) [inline http/task_service.go]: I think this should be `rid`

File: http/task_service.go [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+
+type getRunsRequest struct {
+	filter platform.RunFilter
+}
+
+func decodeGetRunsRequest(ctx context.Context, r *http.Request) (*getRunsRequest, error) {
+	qp := r.URL.Query()
+	req := &getRunsRequest{}
+
+	if id := qp.Get("task"); id != "" {
```
Comment (goller) [inline http/task_service.go]: should this be tid?

File: http/task_service.go [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+		req.filter.Limit = i
+	}
+
+	if time := qp.Get("afterTime"); time != "" {
+		// TODO (jm): verify format is correct, once we've decided on a format
+		req.filter.AfterTime = time
+	}
+
+	if time := qp.Get("beforeTime"); time != "" {
+		// TODO (jm): verify format is correct, once we've decided on a format
```
Comment (goller) [inline http/task_service.go]: I think this should be `rfc3339`

File: task.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+import "context"
+
+// Task is a task. 🎊
+type Task struct {
+	ID     ID     `json:"id,omitempty"`
+	Name   string `json:"name"`
+	Status string `json:"status"`
+	Owner  User   `json:"owner"`
+	IFQL   string `json:"ifql"`
```
Comment (goller) [inline task.go]: probably `query` now (and in the swagger)

File: task.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	// Removes a task by ID and purges all associated data and scheduled runs
+	DeleteTask(ctx context.Context, id ID) error
+
+	// Returns logs for a run.
+	FindLogs(ctx context.Context, filter LogFilter) ([]*Log, int, error)
+
+	// Returns a list of runs that match a filter and the total count of returned runs.
+	FindRuns(ctx context.Context, filter RunFilter) ([]*Run, int, error)
+
+	FindRunByID(ctx context.Context, id ID) (*Run, error)
```
Comment (goller) [inline task.go]: need a comment here

File: http/task_service.go (hunk 4 above)
Comment (jademcgough) [reply http/task_service.go]: yeah, this is what I was thinking... `rfc3339` just feels weird as an http get param for some reason, but I'll probably just use it everywhere

File: task.go (hunk 5 above)
Comment (jademcgough) [reply task.go]: I thought Paul or someone pushed back against using `query`. I can change it though.

File: http/task_service.go (hunk 1 above)
Comment (jademcgough) [reply http/task_service.go]: good catch... this should be pulled from the path

File: http/task_service.go (hunk 2 above)
Comment (goller) [reply http/task_service.go]: is the difference `/tasks/{tid}?run=1` vs `/tasks/{tid}/runs/{rid}` ? 

if so, I would rather scope in the path.

File: task.go (hunk 5 above)
Comment (goller) [reply task.go]: I think it should be query and in the future we can support more dialects with a `type` parameter.

File: http/swagger.yml [hunk 7]
```diff
@@ -688,16 +727,16 @@ components:
           ]
         owner:
           $ref: "#/components/schemas/User"
-        ifql:
-          description: The ifql to run for this task.
+        query:
```
Comment (mark-rushakoff) [inline http/swagger.yml]: We're using the noun `script` internally to tasks. I'd prefer to see script here for consistency. Other considered words included query, code, and program.

File: task.go [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+	// Returns a single run
+	FindRunByID(ctx context.Context, id ID) (*Run, error)
+
+	// Creates and returns a new run (which is a retry of another run)
+	RetryRun(ctx context.Context, id ID) (*Run, error)
+}
+
+// TaskUpdate represents updates to a task
+type TaskUpdate struct {
```
Comment (mark-rushakoff) [inline task.go]: I believe the only thing that can be updated on a task is the script. The name and other details should be specified in the options: https://github.com/influxdata/platform/issues/24

I don't think we have official language support for those options yet, so the workaround for now in tasks is a special comment at start of the script like `//! every=5s;name=request metrics rollup`.

File: http/swagger.yml (hunk 7 above)
Comment (jademcgough) [reply http/swagger.yml]: done

File: task.go (hunk 8 above)
Comment (jademcgough) [reply task.go]: 👍

**Pull request merged**